RUMM-581 Add doc file for each extension package (#315)
DataDog/dd-sdk-android: docs/datadog_ktx_module.md

> # Datadog Kotlin Extensions package

Reviewer: ```# Kotlin Extensions```

Also, instead of `datadog_ktx_module.md` rename this file `kotlin_extensions.md`

> Provides useful Kotlin extensions on top of our core `dd-sdk-android` SDK.

Reviewer: > Take advantage of the Kotlin syntax to improve your usage of the `dd-sdk-android` library.

(I'm trying here to stay consistent with the tone used in the existing docs)

> **Note**: This package is just an extension of our main package so you should add both dependencies into your gradle file.

Reviewer 2: ```suggestion
**Note**: This package is an extension of the main package, so add both dependencies into your gradle file.
```

## Setup
```conf
repositories {
    maven { url "https://dl.bintray.com/datadog/datadog-maven" }
}

dependencies {
    implementation "com.datadoghq:dd-sdk-android:x.x.x"
    implementation "com.datadoghq:dd-sdk-android-ktx:x.x.x"
}
```

### Extensions

> 1. Span extension methods:

Reviewer: Here you're just listing sample code, but I think we do need a couple of lines to explain what it does.

```kotlin

span.setError(throwable)

// or for setting a message
span.setError(message)

```

```kotlin

> // will automatically create a span around the provided lambda

Reviewer 2: ```suggestion
// Automatically creates a span around the provided lambda
```

withinSpan(spanName, parentSpan){

>   // your code here

Reviewer 2: ```suggestion
  // Your code here
```

}

```

2. Extensions for OkHttp request builder:

```kotlin

>   // if you are using our TracingInterceptor for OkHttp request, it will add the provided parentSpan as the parent

Reviewer 2: ```suggestion
  // If you are using the Datadog TracingInterceptor for OkHttp request, this adds the provided parentSpan as the parent
```

>   // of the span we automatically create around the request.

Reviewer 3: It reads a bit convoluted, how about : 
if you are using our TracingInterceptor for OkHttp request, it will add the provided parentSpan as the parent of the span created around the request.

Author: Yup you are right...will change that
